src/test/java/org/wildfly/security/credential/store/KeystorePasswordStoreTest.java

```diff
@@ -132,6 +139,162 @@ public static void remove() {
+            new CredentialStore.CredentialSourceProtectionParameter(
+                IdentityCredentials.NONE.withCredential(new PasswordCredential(
+                    ClearPassword.createRaw(ClearPassword.ALGORITHM_CLEAR, "secret_store_DELETE".toCharArray())))));
+
+        cs.store(passwordAlias1, createCredentialFromPassword(password1));
+        cs.store(passwordAlias2, createCredentialFromPassword(password2));
+
+        Assert.assertArrayEquals(password1, getPasswordFromCredential(cs.retrieve(passwordAlias1, PasswordCredential.class)));
+
+        if (!ks.delete()) {
+            Assert.fail("KeyStore [" + ks.getAbsolutePath() + "] delete fail");
+        }
+
+        Assert.assertArrayEquals(password1, getPasswordFromCredential(cs.retrieve(passwordAlias1, PasswordCredential.class)));
+        // load new entry (in memory)
+        Assert.assertArrayEquals(password2, getPasswordFromCredential(cs.retrieve(passwordAlias2, PasswordCredential.class)));
+
+        cs.store("abc", createCredentialFromPassword(password1));
+        if (!ks.exists()) {
+            Assert.fail("KeyStore [" + ks.getAbsolutePath() + "] must exist yet.");
+        }
+    }
+
+    /**
+     * Credential Store is set to read-only.
+     *
+     * @throws NoSuchAlgorithmException
+     * @throws CredentialStoreException
+     * @throws UnsupportedCredentialTypeException
+     */
+    @Test
+    public void testReadOnly() throws NoSuchAlgorithmException, CredentialStoreException, UnsupportedCredentialTypeException {
+
+        char[] password1 = "secret-password1".toCharArray();
+
+        HashMap<String, String> csAttributes = new HashMap<>();
+
+        csAttributes.put("location", stores.get("TWO"));
+        csAttributes.put("keyStoreType", "JCEKS");
+        csAttributes.put("modifiable", "false");
+
+        String passwordAlias1 = "passAlias_readonly";
+
+        CredentialStore cs = newCredentialStoreInstance();
+        cs.initialize(csAttributes,
+            new CredentialStore.CredentialSourceProtectionParameter(
+                IdentityCredentials.NONE.withCredential(new PasswordCredential(
+                    ClearPassword.createRaw(ClearPassword.ALGORITHM_CLEAR, "secret_store_TWO".toCharArray())))));
+
+        try {
+            cs.store(passwordAlias1, createCredentialFromPassword(password1));
+            Assert.fail("This Credential Store should be read-only.");
+        } catch (CredentialStoreException e) {
+
+        }
+
+        Assert.assertNull("'" + passwordAlias1 + "' must not be in this Credential Store because is read-only.",
+            cs.retrieve(passwordAlias1, PasswordCredential.class));
+    }
+
+    /**
+     * Credential Store entries must be case insensitive.
+     *
+     * @throws NoSuchAlgorithmException
+     * @throws CredentialStoreException
+     * @throws UnsupportedCredentialTypeException
+     */
+    @Test
+    @Ignore("https://issues.jboss.org/browse/WFLY-7742")
+    public void testCaseInsensitiveAlias()
```

Comment:
For any issue that is reproducible within the Elytron codebase can we please reference an ELY- issue?